src/reactive/runner_manager.py

```diff
@@ -0,0 +1,156 @@
+#  Copyright 2024 Canonical Ltd.
+#  See LICENSE file for licensing details.
+
+"""Module for managing reactive runners."""
+import logging
+import shutil
+
+# All commands run by subprocess are secure.
+import subprocess  # nosec
+from dataclasses import dataclass
+from pathlib import Path
+
+from utilities import secure_run_subprocess
+
+logger = logging.getLogger(__name__)
+
+MQ_URI_ENV_VAR = "MQ_URI"
+REACTIVE_RUNNER_LOG_DIR = Path("/var/log/reactive_runner")
+REACTIVE_RUNNER_SCRIPT_FILE = "scripts/reactive_runner.py"
+REACTIVE_RUNNER_TIMEOUT_STR = "1h"
+PYTHON_BIN = "/usr/bin/python3"
+PS_COMMAND_LINE_LIST = ["ps", "axo", "cmd"]
+TIMEOUT_COMMAND = "/usr/bin/timeout"
+UBUNTU_USER = "ubuntu"
+
+
+class ReactiveRunnerError(Exception):
+    """Raised when a reactive runner error occurs."""
+
+
+@dataclass
+class ReactiveRunnerConfig:
+    """Configuration for spawning a reactive runner.
+
+    Attributes:
+        mq_uri: The message queue URI.
+        queue_name: The name of the queue.
+    """
+
+    mq_uri: str
+    queue_name: str
+
+
+def reconcile(quantity: int, config: ReactiveRunnerConfig) -> int:
+    """Spawn a runner reactively.
+
+    Args:
+        quantity: The number of runners to spawn.
+        config: The configuration for the reactive runner.
+
+    Raises a ReactiveRunnerError if the runner fails to spawn.
+
+    Returns:
+        The number of runners spawned.
+    """
+    actual_quantity = _determine_current_quantity()
+    logger.info("Actual quantity of reactive runner processes: %s", actual_quantity)
+    actual_delta = delta = quantity - actual_quantity
+    if delta > 0:
+        logger.info("Will spawn %d new reactive runner processes", delta)
+        _setup_logging()
+        for _ in range(delta):
+            try:
+                _spawn_runner(config)
+            except _SpawnError:
+                logger.exception("Failed to spawn a new reactive runner process")
+        actual_quantity_after_spawning = _determine_current_quantity()
+        actual_delta = actual_quantity_after_spawning - actual_quantity
```

Comment:
Up to you, perhaps just return `delta`

Reply:
I removed the `actual_quantity_after_spawning` and now just count the number of spawned runners (if there is an exception, I increment the counter) and remove that number.